api/api.go

```diff
@@ -765,48 +759,80 @@ func (api *Server) GetActionByActionHash(h hash.Hash256) (action.SealedEnvelope,
 
 // GetEvmTransfersByActionHash returns evm transfers by action hash
 func (api *Server) GetEvmTransfersByActionHash(ctx context.Context, in *iotexapi.GetEvmTransfersByActionHashRequest) (*iotexapi.GetEvmTransfersByActionHashResponse, error) {
-	if !api.hasActionIndex || api.systemLogIndexer == nil {
-		return nil, status.Error(codes.Unavailable, "evm transfer index not supported")
+	return nil, status.Error(codes.Unimplemented, "evm transfer index is deprecated, call GetSystemLogByActionHash instead")
+}
+
+// GetEvmTransfersByBlockHeight returns evm transfers by block height
+func (api *Server) GetEvmTransfersByBlockHeight(ctx context.Context, in *iotexapi.GetEvmTransfersByBlockHeightRequest) (*iotexapi.GetEvmTransfersByBlockHeightResponse, error) {
+	return nil, status.Error(codes.Unimplemented, "evm transfer index is deprecated, call GetSystemLogByBlockHeight instead")
+}
+
+// GetImplicitTransferLogByActionHash returns implict transfer log by action hash
+func (api *Server) GetImplicitTransferLogByActionHash(
+	ctx context.Context,
+	in *iotexapi.GetImplicitTransferLogByActionHashRequest) (*iotexapi.GetImplicitTransferLogByActionHashResponse, error) {
+	if !api.hasActionIndex || api.indexer == nil {
```

Comment:
is this check still need?

Reply:
yes, by action hash first needs to get the action index (which is the block height), then get log by height and match against action hash.
w/o action index it won't be to able to know the height, and cannot proceed

Reply:
I see,that's why GetImplicitTransferLogByBlockHeight does not need this check